activerecord/test/cases/primary_keys_test.rb

```diff
@@ -229,7 +229,7 @@ def test_any_type_primary_key
     assert_equal "code", Barcode.primary_key
 
     column = Barcode.column_for_attribute(Barcode.primary_key)
-    assert_not column.null unless current_adapter?(:SQLite3Adapter)
+    assert_not column.null
```

Comment:
Added the test first at #23959.
This PR fixes the wrong behavior for sqlite3 adapter.